core/src/main/java/com/google/cloud/sql/core/SslSocketFactory.java

```diff
@@ -615,21 +645,23 @@ public long getLastFailureMillis() {
   }
 
   private static class InstanceSslInfo {
-    private final String instanceIpAddress;
+    private final HashMap<String, String> instanceIpAddresses = new HashMap<>();
     private final X509Certificate ephemeralCertificate;
     private final SSLSocketFactory sslSocketFactory;
 
     InstanceSslInfo(
-        String instanceIpAddress,
         X509Certificate ephemeralCertificate,
         SSLSocketFactory sslSocketFactory) {
-      this.instanceIpAddress = instanceIpAddress;
       this.ephemeralCertificate = ephemeralCertificate;
       this.sslSocketFactory = sslSocketFactory;
     }
 
-    public String getInstanceIpAddress() {
-      return instanceIpAddress;
+    public void setInstanceIpAddress(String type, String ipAddress) {
+      instanceIpAddresses.put(type == null ? "" : type.toUpperCase(), ipAddress);
+    }
+
+    public String getInstanceIpAddress(String type) {
+      return instanceIpAddresses.get(type.toUpperCase());
```

Comment:
Shouldn't type already be UpperCase by this point?

Reply:
I think this is just meant to be defensive programming, so at least within the InstanceSslInfo class, the get and put treat the type as case insensitive, but that they do so by canonicalizing into upper case is an implementation detail.